src/kafka/connection/tcp_connection.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#include "tcp_connection.hh"
+
+namespace seastar {
+
+namespace kafka {
+
+auto timeout_end(uint32_t timeout_ms) {
+    return std::chrono::steady_clock::now() + std::chrono::milliseconds(timeout_ms);
+}
+
+future<lw_shared_ptr<tcp_connection>> tcp_connection::connect(const std::string& host, uint16_t port,
```

Comment:
Why this method can't return just tcp_connection and needs a sharde_ptr?

Reply:
We want to avoid copying it around

Reply:
We don't need to copy it - we can move it around.